File: content-solutions/heap/last-stone.java
import java.util.*;

class Solution {
    public int solution(int[] stones) {
        PriorityQueue<Integer> h = new PriorityQueue<>(Collections.reverseOrder());
        for (int s : stones) h.offer(s);
        while (h.size() >= 2) {
            int a = h.poll(), b = h.poll();
            if (a != b) h.offer(a - b);
        }
        return h.isEmpty() ? 0 : h.peek();
    }
}

File: content-solutions/heap/last-stone.py
import heapq


def solution(stones):
    h = [-s for s in stones]
    heapq.heapify(h)
    while len(h) >= 2:
        a = -heapq.heappop(h)
        b = -heapq.heappop(h)
        if a != b:
            heapq.heappush(h, -(a - b))
    return -h[0] if h else 0
